core/modules/system/system.admin.inc

```diff
@@ -123,6 +126,7 @@ function system_themes_page() {
     }
     $admin_theme_options[$theme->name] = $theme->info['name'];
     $theme->is_default = ($theme->name == $theme_default);
+    $theme->is_admin = ($theme->name == $admin_theme);
```

Comment:
The way this is implemented these properties are available only within the context of the themes listing page. I would like us to move these into `system_rebuild_theme_data()`, so that they become available info on themes in general. Would this be an API change?

Reply:
I think moving it into `list_themes()` would be appropriate. It would be an API expansion to do so, since these properties would be made available in more places than they had been previously. That's fine to do in our 1.x release. In our docblock for `list_themes()` we'd just have to include documentation on the new properties and include `@since` sections for the new properties.

Then instead of this function (`system_themes_page()`) calling `$themes = system_rebuild_theme_data();`, it should pull from `list_themes()` instead so it doesn't have to duplicate the work.